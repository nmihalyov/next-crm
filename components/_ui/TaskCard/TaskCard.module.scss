@import '../../../styles//variables';

.card {
  padding: 20px 0;
  &:not(:last-child) {
    border-bottom: 1px solid transparentize($primary, .9);
  }
}

.title {
  transition: opacity .3s ease;
  &Done {
    opacity: 0.5;
  }
}