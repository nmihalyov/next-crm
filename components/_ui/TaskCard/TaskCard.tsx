import { FC, memo } from 'react';
import type { Task } from '../../../types/task';
import { Modal, Row, Button, Typography, Checkbox } from 'antd';
import { DeleteOutlined } from '@ant-design/icons';

import styles from './TaskCard.module.scss';

const TaskCard: FC<Task & {
  onPatch(id: number, completed: boolean): void,
  onRemove(id: number): void,
}> = props => {
  const { id, title, completed, onPatch, onRemove } = props;
  const onChange = () => onPatch(id, !completed);
  const removeHandler = (): void => {
    Modal.confirm({
      title: 'Are you sure you want to delete this task?',
      okText: 'Yes',
      okType: 'danger',
      cancelText: 'No',
      onOk() {
        onRemove(id);
      }
    });
  };

  return (
    <Row className={styles.card} justify="space-between" align="middle">
      <Checkbox checked={completed} onChange={onChange}>
        <Typography.Text className={`${styles.title} ${completed ? styles.titleDone : ''}`}>{title}</Typography.Text>
      </Checkbox>
      <Button
        danger
        ghost
        type="primary"
        icon={<DeleteOutlined />}
        onClick={removeHandler} />
    </Row>
  );
};

export default memo(TaskCard);