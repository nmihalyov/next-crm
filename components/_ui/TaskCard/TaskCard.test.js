import TaskCard from './TaskCard';

describe('TaskCard:', () => {
  const setUp = props => shallow(<TaskCard {...props} />);

  it('renders TaskCard', () => {
    const props = {
      userId: 1,
      id: 1,
      title: 'Task title',
      completed: false,
      onPatch: () => {},
      onRemove: () => {}
    };
    const component = setUp(props);

    expect(component).toMatchSnapshot();
  });

  it('title has completed class name if completed prop is true', () => {
    const component = setUp({completed: true});
    const title = component.find('Text');

    expect(title.hasClass(/titleDone/)).toBe(true);
  });

  it('calls onPatch with correct arguments after change', () => {
    const onPatch = jest.fn();
    const id = 1;
    const completed = true;
    const component = setUp({onPatch, id, completed});

    component.find('Checkbox').simulate('change');
    expect(onPatch).toHaveBeenCalledWith(id, !completed);
  });
});