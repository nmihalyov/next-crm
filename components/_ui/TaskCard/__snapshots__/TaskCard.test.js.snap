// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskCard: renders TaskCard 1`] = `
<Row
  align="middle"
  className="card"
  justify="space-between"
>
  <Checkbox
    checked={false}
    onChange={[Function]}
  >
    <Text
      className="title "
    >
      Task title
    </Text>
  </Checkbox>
  <Button
    danger={true}
    ghost={true}
    icon={<ForwardRef(DeleteOutlined) />}
    onClick={[Function]}
    type="primary"
  />
</Row>
`;
